File: ABC/ABC135/B/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {

  int N;
  cin >> N;

  int count = 0;

  for (int i = 1; i <= N; i++) {
    int p;
    cin >> p;
    if (p != i) {
      count++;
    } 
  }

  if (count <= 2) {
    cout << "YES" << endl;
  } else {
    cout << "NO" << endl;
  }

}

File: ABC/ABC135/B/main.java
import java.util.*;
public class Main {

  public static void main (String[] args) {
	
    Scanner sc = new Scanner(System.in);
    
    int N = sc.nextInt();
    
    Integer[] p = new Integer[N];
    
    for (int i = 0; i < N; i++) {
      p[i] = sc.nextInt();
    }

    sc.close();

    Integer[] p_ = p.clone();

    Arrays.sort(p);

    int count = 0;

    for (int i = 0; i < N; i++){
      if (p[i] != p_[i]){
        count++;
      }
    }

    if (count == 0 || count == 2){
      System.out.println("YES");
    } else {
      System.out.println("NO");
    }
    
  }

}